### ISSUE TITLE
Accessibility Issues

### ISSUE THREAD

**Issue by jeffkamo** [mobify/pinny#29]
**Keyboard Accessibility**
- [ ] After opening pinny with keyboard, focus stays on the element that opened pinny (in the example page, that would be the sheet up/right/bottom/left/center buttons)
- [ ] After opening pinny, navigating with the keyboard should ONLY navigate inside the pinny - you should NOT be able to interact with anything outside of the modal (example jQuery implementation can be found [here](http://accessibility.oit.ncsu.edu/blog/2013/09/13/the-incredible-accessible-modal-dialog/))
- [ ] `Esc` key should be mapped to closing the modal

**Screen Reader Accessibility**
- [ ] Interactivity with the page main content must be disabled. This can be achieved by adding `aria-hidden="true"` to whatever is wrapping the main content (the example pages don't have any identifiers on the main content div!!)

``` html
<!-- Before pinny is opened... -->
<div class="t-main" aria-hidden="false"></div>
<section class="pinny" aria-hidden="true"></section>

<!-- After pinny is opened... -->
<div class="t-main" aria-hidden="true"></div>
<section class="pinny pinny--is-open" aria-hidden="false"></section>
```
- [ ] A (perhaps overridable) description div should be included always - it's purpose is to describe the modal and how to interact with it. Implementation on the rendered html would require the description div having a unique id, and that id referred to on the pinny in a `aria-describedby` attribute

``` html
<section class="pinny" describedby="pinnyDescription" role="dialogue">
    <div id="pinnyDescription" class="visually-hidden"><!-- description text --></div>
    <!-- .pinny__header -->
    <!-- .pinny__content -->
</section>
```
- [ ] Pinny heading should have a unique id and identified on the pinny in a `aria-labelledby` attribute

``` html
<section class="pinny" labelledby="pinnyLabel" role="dialogue">
    <!-- description block -->
    <header class="pinny__header">
        <h1 id="pinnyLabel" class="pinny__title"></h1>
    </header>
    <!-- .pinny__content -->
</section>
```
- [ ] There should be an inner wrapper with the `role="document"` property. See the reasons in [this blog post](http://accessibility.oit.ncsu.edu/blog/2014/01/02/incredible-accessible-modal-window-version-2/) and in point 8 of the blog post's [working example](http://accessibility.oit.ncsu.edu/training/aria/modal-window/version-2/).

``` html
<section class="pinny" labelledby="pinnyLabel" role="dialogue">
    <div role="document">
        <!-- description block -->
        <!-- .pinny__header -->
        <!-- .pinny__content -->
    </div>
</section>
```

**W3C Recommendations**
- [ ] The first focusable item in the modal dialog should receive the keyboard focus.
- [ ] The window behind the modal dialog should not be allowed to be clicked on
- [ ] The modal dialog must trap the keyboard focus inside the modal dialog so the user can’t accidentally interact with the window behind the modal dialog.
  - [ ] When the user is on the last focusable item and presses Tab, the user should be taken to the first focusable item in the modal dialog.
  - [ ] When the user is on the first  focusable item and presses Shift-Tab, the user should be taken to the last focusable item in the modal dialog.
- [ ] The position of the keyboard focus before the modal window opens must be saved, and the focus must be restored to this location after the modal dialog closes.

---

sources
- http://accessibility.oit.ncsu.edu/blog/2013/09/13/the-incredible-accessible-modal-dialog/
- http://accessibility.oit.ncsu.edu/blog/2014/01/02/incredible-accessible-modal-window-version-2/
- http://accessibility.oit.ncsu.edu/training/aria/modal-window/version-2/
